panic: runtime error: slice bounds out of range sometimes occur on other nodes when a node is taken offline

## Expected Behavior
When a node is taken offline, the remaining nodes are expected to keep running.

## Current Behavior
When a node is taken offline, other nodes sometimes panics and restart.

Two example panics from two other nodes during the restart of a third node:

```
panic: runtime error: slice bounds out of range [36:32]

goroutine 62354072 [running]:
github.com/minio/highwayhash.(*digest).Write(0xc0011de000, {0xc02f009e20, 0x6, 0x100000002d9437e})
        github.com/minio/highwayhash@v1.0.2/highwayhash.go:171 +0x265
github.com/minio/minio/internal/logger.hashString({0xc02808b6e5, 0x7})
        github.com/minio/minio/internal/logger/logger.go:267 +0xbf
github.com/minio/minio/internal/logger.logIf({0x4478890, 0xc01c87edc0}, {0x4420ea0, 0xc02d6858d0}, {0x0, 0x40d187, 0x10})
        github.com/minio/minio/internal/logger/logger.go:374 +0x557
github.com/minio/minio/internal/logger.LogIf({0x4478890, 0xc01c87edc0}, {0x4420ea0, 0xc02d6858d0}, {0x0, 0x0, 0x0})
        github.com/minio/minio/internal/logger/logger.go:310 +0x14f
github.com/minio/minio/cmd.(*storageRESTClient).WalkDir(0x474d25, {0x4478890, 0xc01c87edc0}, {{0xc02808b6e5, 0x6}, {0xc01a4e79c8, 0x17}, 0x1, 0x0, {0x0, ...}, ...}, ...)
        github.com/minio/minio/cmd/metacache-walk.go:329 +0x5b2
github.com/minio/minio/cmd.listPathRaw.func2()
        github.com/minio/minio/cmd/metacache-set.go:818 +0x178
created by github.com/minio/minio/cmd.listPathRaw
        github.com/minio/minio/cmd/metacache-set.go:813 +0x21d
```

```
panic: runtime error: slice bounds out of range [:40] with length 32

goroutine 62650594 [running]:
github.com/minio/highwayhash.hashBuffer(0x10100c028f63478, 0x7f47756eaf08, 0x0)
        github.com/minio/highwayhash@v1.0.2/highwayhash.go:215 +0x216
github.com/minio/highwayhash.(*digest).Sum(0xc001254000, {0x0, 0x0, 0x0})
        github.com/minio/highwayhash@v1.0.2/highwayhash.go:189 +0x93
github.com/minio/minio/internal/logger.hashString({0xc03ae7f2d5, 0x7})
        github.com/minio/minio/internal/logger/logger.go:268 +0xda
github.com/minio/minio/internal/logger.logIf({0x4478938, 0xc03b4ff440}, {0x4420ee0, 0xc03b74d760}, {0x0, 0x0, 0xc02732d440})
        github.com/minio/minio/internal/logger/logger.go:374 +0x557
github.com/minio/minio/internal/logger.LogIf({0x4478938, 0xc03b4ff440}, {0x4420ee0, 0xc03b74d760}, {0x0, 0x0, 0x0})
        github.com/minio/minio/internal/logger/logger.go:310 +0x14f
github.com/minio/minio/internal/logger.(*logOnceType).logOnceIf(0xc0001c1260, {0x4478938, 0xc03b4ff440}, {0x4420ee0, 0xc03b74d760}, {0x2831840, 0xc03bba96c0}, {0x0, 0x0, 0x0})
        github.com/minio/minio/internal/logger/logonce.go:54 +0x285
github.com/minio/minio/internal/logger.LogOnceIf({0x4478938, 0xc03b4ff440}, {0x4420ee0, 0xc03b74d760}, {0x2831840, 0xc03bba96c0}, {0x0, 0x0, 0x0})
        github.com/minio/minio/internal/logger/logonce.go:94 +0x1b5
github.com/minio/minio/cmd.readAllFileInfo.func1()
        github.com/minio/minio/cmd/erasure-metadata-utils.go:139 +0x49d
github.com/minio/minio/internal/sync/errgroup.(*Group).Go.func1()
        github.com/minio/minio/internal/sync/errgroup/errgroup.go:123 +0x1cb
created by github.com/minio/minio/internal/sync/errgroup.(*Group).Go
        github.com/minio/minio/internal/sync/errgroup/errgroup.go:105 +0xa7
```


## Steps to Reproduce (for bugs)

I cannot reproduce this reliably other than it usually happens during the restart of a node (pod). Possibly while there are multiple read requests in progress.

## Context
Maintenance and/or draining pods from one Kubernetes node.

## Your Environment
* Version used (`minio --version`): 2021-11-09T03:21:45Z
* Server setup and configuration: 5 nodes, 1 disk each, Kubernetes/GKE

--- a/internal/logger/logger.go
+++ b/internal/logger/logger.go
@@ -23,7 +23,6 @@ import (
 	"errors"
 	"fmt"
 	"go/build"
-	"hash"
 	"net/http"
 	"path/filepath"
 	"reflect"
@@ -39,8 +38,6 @@ import (
 var (
 	// HighwayHash key for logging in anonymous mode
 	magicHighwayHash256Key = []byte("\x4b\xe7\x34\xfa\x8e\x23\x8a\xcd\x26\x3e\x83\xe6\xbb\x96\x85\x52\x04\x0f\x93\x5d\xa3\x9f\x44\x14\x97\xe0\x9d\x13\x22\xde\x36\xa0")
-	// HighwayHash hasher for logging in anonymous mode
-	loggerHighwayHasher hash.Hash
 )
 
 // Disable disables all logging, false by default. (used for "go test")
@@ -207,8 +204,6 @@ func Init(goPath string, goRoot string) {
 	// paths like "{GOROOT}/src/github.com/minio/minio"
 	// and "{GOPATH}/src/github.com/minio/minio"
 	trimStrings = append(trimStrings, filepath.Join("github.com", "minio", "minio")+string(filepath.Separator))
-
-	loggerHighwayHasher, _ = highwayhash.New(magicHighwayHash256Key) // New will never return error since key is 256 bit
 }
 
 func trimTrace(f string) string {
@@ -263,10 +258,9 @@ func getTrace(traceLevel int) []string {
 
 // Return the highway hash of the passed string
 func hashString(input string) string {
-	defer loggerHighwayHasher.Reset()
-	loggerHighwayHasher.Write([]byte(input))
-	checksum := loggerHighwayHasher.Sum(nil)
-	return hex.EncodeToString(checksum)
+	hh, _ := highwayhash.New(magicHighwayHash256Key)
+	hh.Write([]byte(input))
+	return hex.EncodeToString(hh.Sum(nil))
 }
 
 // Kind specifies the kind of error log

